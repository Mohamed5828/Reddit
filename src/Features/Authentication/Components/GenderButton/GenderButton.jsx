import React from "react";
import { BaseButton } from "./GenderButton.styled";

/**
 * GenderButton component that is used in AfterSignUp modals
 * @param {string} children The text of the button
 * @param {Array} otherProps Props to know the shape of the button
 * @returns {React.Component}  GenderButton component that is used in AfterSignUp modals
 */
const GenderButton = ({ children, ...otherProps }) => {
  return <BaseButton {...otherProps}>{children}</BaseButton>;
};

export default GenderButton;
