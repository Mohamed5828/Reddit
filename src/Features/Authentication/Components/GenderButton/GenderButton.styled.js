import styled from "styled-components";

const fontFamily = "'IBM Plex Sans', sans-serif;";

export const BaseButton = styled.button`
  line-height: 40px;
  background-color: ${({ theme }) => theme.color.mainInputBackGroundColor};

  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-family: Noto Sans, sans-serif;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: unset;
  border-radius: ${(p) => (p.page ? "5px" : "999px")};
  color: black;
  height: 40px;
  padding: 0 16px;
  font-family: ${fontFamily};
  text-transform: unset;
  margin-top: 8px;
  width: 100%;

  

  cursor: pointer;

  &:hover {
    background-color: white;
    border: 1px solid black;
  }
`;
