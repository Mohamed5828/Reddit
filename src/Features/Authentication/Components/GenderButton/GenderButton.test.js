import { shallow } from "enzyme";

import GenderButton from "./GenderButton";

describe("GenderButton Component", () => {
  it("this is a test for GenderButton Component", () => {
    expect(shallow(<GenderButton />)).toMatchSnapshot();
  });
});
