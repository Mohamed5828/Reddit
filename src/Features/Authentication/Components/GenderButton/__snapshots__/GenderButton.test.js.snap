// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenderButton Component this is a test for GenderButton Component 1`] = `<styled.button />`;
